gateway-ha/src/test/java/io/trino/gateway/ha/persistence/BaseTestDatabaseMigrations.java

```diff
@@ -0,0 +1,102 @@
+import org.jdbi.v3.core.Jdbi;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.Isolated;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+import static org.junit.jupiter.api.parallel.ExecutionMode.SAME_THREAD;
+
+@TestInstance(PER_CLASS)
+@Execution(SAME_THREAD)
+@Isolated
+public abstract class BaseTestDatabaseMigrations
+{
+    protected final JdbcDatabaseContainer<?> container = startContainer();
+    protected final Jdbi jdbi = Jdbi.create(container.getJdbcUrl(), container.getUsername(), container.getPassword());
+
+    protected abstract JdbcDatabaseContainer<?> startContainer();
+
+    protected abstract String getDriver();
+
+    @AfterAll
+    public final void close()
+    {
+        container.close();
+    }
+
+    @Test
+    public void testMigrationWithEmptyDatabase()
+    {
+        DataStoreConfiguration config = new DataStoreConfiguration(
+                container.getJdbcUrl(),
+                container.getUsername(),
+                container.getPassword(),
+                getDriver(),
+                4,
+                true);
+        FlywayMigration.migrate(config);
+        verifyGatewaySchema(0);
+
+        dropAllTables();
+    }
+
+    protected void verifyGatewaySchema(int expectedPropertiesCount)
+    {
+        verifyResultSetCount("SELECT name FROM gateway_backend", 0);
+        verifyResultSetCount("SELECT query_id FROM query_history", 0);
+        verifyResultSetCount("SELECT name FROM resource_groups_global_properties", expectedPropertiesCount);
+        verifyResultSetCount("SELECT name FROM resource_groups", 0);
+        verifyResultSetCount("SELECT user_regex FROM selectors", 0);
+        verifyResultSetCount("SELECT environment FROM exact_match_source_selectors", 0);
+    }
+
+    private void verifyResultSetCount(String sql, int expectedCount)
+    {
+        List<String> results = jdbi.withHandle(handle ->
+                handle.createQuery(sql).mapTo(String.class).list());
+        assertThat(results).hasSize(expectedCount);
+    }
+
+    protected void dropAllTables()
+    {
+        String gatewayBackendTable = "DROP TABLE IF EXISTS gateway_backend";
+        String queryHistoryTable = "DROP TABLE IF EXISTS query_history";
+        String propertiesTable = "DROP TABLE IF EXISTS resource_groups_global_properties";
+        String resourceGroupsTable = "DROP TABLE IF EXISTS resource_groups";
+        String selectorsTable = "DROP TABLE IF EXISTS selectors";
+        String exactMatchTable = "DROP TABLE IF EXISTS exact_match_source_selectors";
+        String flywayHistoryTable = "DROP TABLE IF EXISTS flyway_schema_history";
+        Handle jdbiHandle = jdbi.open();
```

Comment:
Can we add a check that the schema is now empty? A query against `information_schema` should work for both pg and mysql. This will ensure that if someone adds a table to the schema, they remember to update this method